masabot/commands/karma.py

```diff
@@ -90,18 +92,41 @@ def set_state(self, state):
 				msg = "Buzzkill mode enabled;"
 				msg += " karma change greater than " + str(self._buzzkill_limit) + " not allowed"
 			else:
-				msg = self.add_user_karma(user, amount)
+				if hasattr(context.source, "server"): 
```

Comment:
As best as I can tell, there are only two instances where there could ever not be a server attached to the source of the message:
1.) In a DM with MasaBot, where Karma should not be added for other users.
2.) In a group chat (not attached to a server).

Either way, we should try to avoid using `hasattr`, as meta-programming can lead to some tricky situations down the line. We can already check if we are in a PM with the `is_pm` attribute of `BotContext`, but for checking group chat, this may be the only way for now.

For readability and future maintainability, instead of using `hasattr` here, I think we should consider adding a new `is_group_chat` attribute to `BotContext` and using that, unless there was another reason for this check to be here.